plugins/nodes/src/js/columns/NodesTableDiskColumn.tsx

```diff
@@ -1,33 +1,41 @@
 import * as React from "react";
-import { compareValues } from "#PLUGINS/nodes/src/js/utils/compareValues";
+import sort from "array-sort";
 import Node from "#SRC/js/structs/Node";
 // TODO: DCOS-39079
 // import { IWidthArgs as WidthArgs } from "@dcos/ui-kit/packages/table/components/Column";
 import { IWidthArgs as WidthArgs } from "#PLUGINS/nodes/src/js/types/IWidthArgs";
 import { SortDirection } from "plugins/nodes/src/js/types/SortDirection";
 import { TextCell } from "@dcos/ui-kit";
 
+function getDiskUsage(data: Node) {
+  return data.getUsageStats("disk").percentage;
+}
+
 export function diskRenderer(data: Node): React.ReactNode {
   return (
     <TextCell>
-      <span>{data.getUsageStats("disk").percentage}%</span>
+      <span>{getDiskUsage(data)}%</span>
     </TextCell>
   );
 }
 
+function compareNodesByDiskUsage(a: Node, b: Node): number {
+  return getDiskUsage(a) - getDiskUsage(b);
+}
+
+function compareNodesByHostname(a: Node, b: Node): number {
+  return a
+    .getHostName()
+    .toLowerCase()
+    .localeCompare(b.getHostName().toLowerCase());
+}
+
+const comparators = [compareNodesByDiskUsage, compareNodesByHostname];
 export function diskSorter(data: Node[], sortDirection: SortDirection): Node[] {
-  // current implementation is a rough idea, not sure if it is the best one…
-  const sortedData = data.sort((a, b) =>
-    compareValues(
-      a.get("used_resources").disk.toString(),
-      b.get("used_resources").disk.toString(),
-      a.getHostName().toLowerCase(),
-      b.getHostName().toLowerCase()
-    )
-  );
-  return sortDirection === "ASC" ? sortedData : sortedData.reverse();
+  const reverse = sortDirection !== "ASC";
+  return sort(data, comparators, { reverse });
 }
-export function diskSizer(args: WidthArgs): number {
+export function diskSizer(_args: WidthArgs): number {
```

Comment:
This will conflict with another currently open PR - https://github.com/dcos/dcos-ui/pull/3110/files#diff-1e1caaa9ce248bceffeda8f7749a5de1R33